modules/react-native-status/android/src/main/java/im/status/ethereum/module/JSCJail.java

```diff
@@ -0,0 +1,270 @@
+        }
+
+        String setInterval(JSValue callback, int interval) {
+            return this.scheduleTask(callback, interval, true);
+        }
+
+        private String scheduleTask(final JSValue callback, int interval, final boolean repeatable) {
+            final String id = UUID.randomUUID().toString();
+            final ScheduledExecutorService scheduler =
+                    Executors.newSingleThreadScheduledExecutor();
+
+            timers.put(id, scheduler);
+
+            scheduler.scheduleAtFixedRate
+                    (new Runnable() {
+                        public void run() {
+                            if (!repeatable) {
+                                scheduler.shutdown();
+                                timers.remove(id);
+                            }
+                            callback.toFunction().call();
+                        }
+                    }, interval, interval, TimeUnit.MILLISECONDS);
+
+            return id;
+        }
+
+        void clearInterval(String id) {
+            if (!timers.containsKey(id)) {
+                return;
+            }
+            ScheduledExecutorService scheduler = timers.get(id);
+            scheduler.shutdown();
+            timers.remove(id);
+        }
+
+        void reset() {
+            for (String entry : timers.keySet()) {
+                timers.get(entry).shutdown();
+            }
+            timers.clear();
+        }
+
+    }
+
+
+    private void addHandlers(Cell cell, final String chatId) {
+        JSContext context = cell.context;
+        final Timer timer = cell.timer;
+
+        JSFunction web3send = new JSFunction(context, "web3send") {
+            public String web3send(String payload) {
+                return Statusgo.CallRPC(payload);
+            }
+        };
+        context.property("web3send", web3send);
+
+        JSFunction web3sendAsync = new JSFunction(context, "web3sendAsync") {
+            public void web3sendAsync(final String payload, final JSValue callback) {
+                Thread thread = new Thread() {
+                    @Override
+                    public void run() {
+                        String result = Statusgo.CallRPC(payload);
+                        callback.toFunction().call(null, result);
+                    }
+                };
+
+                thread.start();
+            }
+        };
+        context.property("web3sendAsync", web3sendAsync);
+
+        JSFunction statusSendSignal = new JSFunction(context, "statusSendSignal") {
+            public void statusSendSignal(String data) {
+                JSONObject event = new JSONObject();
+                JSONObject signal = new JSONObject();
+                try {
+                    event.put("chat_id", chatId);
+                    event.put("data", data);
+                    signal.put("type", "jail.signal");
+                    signal.put("event", event);
+                } catch (JSONException e) {
+
```

Comment:
log

Reply:
tbh i wouldn't mind to give these exception an opportunity to be thrown in the runtime, but eh.. ok

Reply:
This would work too

Reply:
No it wouldn't, that's java :D i have to rewrite it to these RuntimeException or something, so let's just have log messages